private/buf/bufgen/internal/bufgenv2/config.go

```diff
@@ -279,18 +287,34 @@ func mergeDisabledFuncs(disabledFuncs []disabledFunc) disabledFunc {
 
 func mergeOverrideFuncs(overrideFuncs []overrideFunc) overrideFunc {
 	// Last override listed wins
```

Comment:
this function implements the rule that:
If a file matches both a rule on `java_package` and a rule on `java_package_prefix` / `java_package_suffix`, the last one wins.
If a file the last java package rule (one of the three) matched by a file is `java_package_suffix`, we will also apply the last seen `java_package_prefix`, and vice versa.